app/configurator/config-form.tsx

```diff
@@ -87,22 +102,29 @@ export const useChartFieldField = ({
             sourceType: state.dataSource.type,
           }
         )
-        .toPromise();
-      const tree = hierarchyData?.dataCubeByIri?.dimensionByIri
-        ?.hierarchy as HierarchyValue[];
-
-      // If the dimension has a hierarchy, we select leaves
-      const leaves = getLeaves(tree);
-
-      dispatch({
-        type: "CHART_FIELD_CHANGED",
-        value: {
-          locale,
-          field,
-          componentIri: dimensionIri,
-          selectedValues: leaves,
-        },
-      });
+        .toPromise()
+        .then((response) => {
+          const tree = response.data?.dataCubeByIri?.dimensionByIri
+            ?.hierarchy as HierarchyValue[];
+          const leaves = getLeaves(tree);
+
+          if (!unmountedRef.current) {
+            dispatch({
+              type: "CHART_FIELD_CHANGED",
+              value: {
+                locale,
+                field,
+                componentIri: dimensionIri,
+                selectedValues: leaves,
+              },
+            });
+          }
+        })
+        .finally(() => {
+          if (!unmountedRef.current) {
+            setFetchingHierarchy(false);
+          }
+        });
```

Comment:
Why don't we use a try/finally here ?

Reply:
Not sure why I've added finally there, I have moved this logic up inside .then call as anyway it is not supposed to fail, even when user switching to another chart in the meantime :)

Reply:
Could we use an `await` then ?

Reply:
Yes, I've changed the code to use `await` :)